### ISSUE TITLE
[Accessibility]: Code Insights Dashboard - Add or remove insights button does not read

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41683]
### Audit type

Screen reader navigation

### Problem description

![PixelSnap 2022-09-15 at 12 50 17](https://user-images.githubusercontent.com/1855233/190475101-758d32d3-8ade-4b01-bd32-846681167e04.png)

When the "Add or remove insights" is disabled, it does not read the button. It just says, "you are currently on a group." It also says to press "ctrl+shift+opt+down arrow" to step into the group, but this is not possible. When you try that, it gives a chime indicating that you can't.

### Expected behavior

When focusing on the button, it should read the text of the button and indicate that the button is disabled.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @joelkw @felixfbecker @vovakulikov

**Comment by vovakulikov** [comment]
@sourcegraph/frontend-platform-devs I think this problem comes from the fact that for disabled button we use a special wrapper element and this probably breaks screen reader flow.

**Comment by felixfbecker** [comment]
I think instead of a wrapper element, we should simply use `aria-disabled="true"` on the button and override `onClick={event => event.preventDefault()}` instead of using the `disabled` attribute. See https://css-tricks.com/making-disabled-buttons-more-inclusive/.

![Image](https://user-images.githubusercontent.com/10532611/200475477-943f05d1-8119-4828-b699-ad9421c8086e.png)

Another problem here is that the tooltip is not announced. I think this is also because of the wrapper element, `aria-describedby` is on the parent element, not on the wrapper `div` that has `tabindex="0"`. Moving all of this onto the `<button>` again should fix this.

**Comment by felixfbecker** [comment]
Duplicate of https://github.com/sourcegraph/sourcegraph/issues/38782
